src/components/signup/SignupState.js

```diff
@@ -87,6 +87,63 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
       log.error('Could not get user location', e)
     }
   }
+
+  const checkWeb3Token = async () => {
+    setLoading(true)
+    const web3Token = await AsyncStorage.getItem('web3Token')
+
+    if (web3Token) {
+      let behaviour = ''
+
+      try {
+        const w3userData = await API.getUserFromW3ByToken(web3Token)
+        const w3user = w3userData.data
+
+        if (w3user.has_wallet) {
+          behaviour = 'goToRecoverScreen'
+        } else {
+          const requestedData = {
+            w3Token: web3Token,
+            email: w3user.email,
+            isEmailConfirmed: true,
+            fullName: w3user.full_name,
+            avatar: w3user.image,
```

Comment:
do we get it in the same format? in the app we save it as base64, you probably get a url back no?